src/recon_buildblock/PLSPrior.cxx

```diff
@@ -258,7 +264,21 @@ PLSPrior<elemT>::
 set_kappa_sptr(const shared_ptr<DiscretisedDensity<3,elemT> >& k)
 { this->kappa_ptr = k; }
 
+//! Set kappa filename
+template <typename elemT> void PLSPrior<elemT>::set_kappa_filename(const std::string& filename)
+{
+  kappa_filename  = filename;
+  this->kappa_ptr = read_from_file<DiscretisedDensity<3,elemT> >(kappa_filename);
+  info(boost::format("Reading kappa data '%1%'") % kappa_filename  );
+}
 
+//! Set anatomical filename
+template <typename elemT> void PLSPrior<elemT>::set_anatomical_filename(const std::string& filename)
+{
+  anatomical_filename  = filename;
+  this->anatomical_sptr = read_from_file<DiscretisedDensity<3,elemT> >(anatomical_filename);
+  info(boost::format("Reading anatomical data '%1%'") % anatomical_filename  );
```

Comment:
do we have a set_anatomical_sptr()? Should there set the filename to zero., same for kappa

Reply:
There are three ways to set the anatomical/kappa images. 
1. parser
2. set_filename
3. set_sptr
The parser will call the set_filename function in post_processing. The set_filename will clear the filename at the end. set_sptr will never alter the filename. So in all three cases, the image will be read and the filename will be cleared.

Reply:
the removal of the filename doesn't look ok to me. A user might be surprised that if they set a filename, it suddenlly isn't valid anymore. This will in particular occur with the `parameter_info` from `ParsingObject`. It seems that `set_sptr` has to clear the filename (as it doesn't know what it is), but `set_filename` keeps it. as the parser will use `set_filename`, everything should still be fine.

Reply:
done.